index.js

```diff
@@ -19,26 +21,44 @@ util.inherits(CpFileError, NestedError);
 
 CpFileError.prototype.name = 'CpFileError';
 
+function statSource(src, progress) {
+	if (!progress.enabled) {
+		return Promise.resolve();
+	}
+	return fsP.stat(src).then(function (stat) {
+		progress.stat = stat;
+	}).catch(function (err) {
+		Promise.reject(new CpFileError('NO_ENTRY', err));
```

Comment:
This isn't doing anything as it's not returned. Needs a regression test for that.

The correct solution is to just `throw` the error.